Assets/MapGenerator.cs

```diff
@@ -0,0 +1,274 @@
+﻿using UnityEngine;
+using System.Collections;
+using System;
+
+public class MapGenerator : MonoBehaviour 
+{
+
+	public int width;
+	private string seed;
+	private bool useRandomSeed = true;
+
+	[Range(0,100)]
+	public int randomCaveSeedPercent = 50;
+	[Range(0, 20)]
+	public int SmoothingIterations = 5;
+
+	public float heightScale = 1.0F;
+	public float xScale = 1.0F;
+	public float yNoise = 0f;
+	public float groundDepth = 50.0f;
+	public float hillDepth = 20.0f;
+	public float skyDepth = 20.0f;
+	public int height;
+
+	public AnimationCurve contourCurve = AnimationCurve.Linear (0f, 0f, 1f, 1f);
+	private float GroundTop;
+	public long heightSum;
+
+
+	[Range(0, 8)]
+	public int BirthLimit = 4;		//if the number of walls around the selected blank point is high enough then fill it in
+	[Range(0, 8)]
+	public int DeathLimit = 5;		//if the number of walls around the selected non-blank point is not too high then kill it
+
+
+	int[,] map;
+	int[,] newMap;
+	//MapCol[] map;
+	int[] Topography;
+
+	void Start() 
+	{
+		GenerateMap();
+	}
+
+	void Update() 
+	{
+		if (Input.GetMouseButtonDown(0)) 
+		{
+			GenerateMap();
+		}
+	}
+
+	void GenerateMap() 
+	{
+		height = (int)(groundDepth + hillDepth + skyDepth);
+		//map = new MapCol[width];
+		map = new int[width,height];
+		Topography = new int[width];
+		heightSum = 0;
+		ClearMap ();
+		//RandomFillMap();
```

Comment:
First comment is a stub of the code that we would like to switch to after splitting out the classes.
Second comment can goaway. You are correct that we will never use RandomFillMap